emily/handler/src/common/error.rs

```diff
@@ -128,21 +97,130 @@ pub enum Error {
+
+    /// This happens when the request to DynamoDB succeeds but does not
+    /// return any values. This happens when the request instructs the
+    /// database to refrain from returning values, so this is likely a
+    /// programming error.
+    #[error("entry in database for deposit request not returned from DynamoDB; {0}")]
+    MissingAttributesDeposit(DepositEntryKey),
+
+    /// This happens when the request to DynamoDB succeeds but does not
+    /// return any values. This happens when the request instructs the
+    /// database to refrain from returning values, so this is likely a
+    /// programming error.
+    #[error("entry in database for withdrawal request not returned from DynamoDB; {0}")]
+    MissingAttributesWithdrawal(WithdrawalEntryKey),
+
+    /// DynamoDB should only contain one entry per withdrawal request ID.
+    ///
+    /// TODO: In case of a re-org, tripple check that we can identify the
+    /// correct withdrawal request if the transaction is replayed.
+    #[error("DynamoDB contained many entries for the given request ID: {0}")]
+    TooManyWithdrawalEntries(u64),
+
+    /// This happens when we fail to decode a base64 encoded string into a
+    /// vector of bytes.
+    #[error("failed to base64 decode the string into bytes; {0}")]
+    Base64Decode(base64::DecodeError),
+
+    /// This is used when trying to get a required value from the
+    /// environment and that operation fails.
+    #[error("could not read the environment variable; {0}")]
+    EnvVariable(#[from] env::VarError),
+
+    /// This occurs when serializing or deserializing an object into or
+    /// from JSON.
+    #[error("{0}")]
+    SerdeJson(#[from] serde_json::Error),
+
+    /// This occurs when converting structs to and from DynamoDB objects.
+    #[error("{0}")]
+    SerdeDynamo(#[from] serde_dynamo::Error),
+
+    /// This happens when attempting to parse an integer from a string that
+    /// has been read from an environment variable.
+    #[error("could not parse the string into an integer; {0}")]
+    EnvParseInt(#[from] std::num::ParseIntError),
+
+    /// This happens when attempting to read an item from DynamoDB.
+    #[error("could not retrieve an item from DynamoDB; {0}")]
+    AwsSdkDynamoDbGetItem(#[from] SdkError<GetItemError>),
+
+    /// This happens when attempting to store an item in DynamoDB. Note
+    /// that precondition checks that occur when putting an item into
+    /// DynamoDB are transformed into the `VersionConflict` error variant.
+    #[error("could not put the item into DynamoDB; {0}")]
+    AwsSdkDynamoDbPutItem(#[source] Box<PutItemError>),
+
+    /// This happens when attempting the "Query" operation in DynamoDB.
+    #[error("could complete Query operation on DynamoDB; {0}")]
+    AwsSdkDynamoDbQuery(#[from] SdkError<QueryError>),
+
+    /// This happens when attempting the "Scan" operation in DynamoDB.
+    #[error("could complete Scan operation on DynamoDB; {0}")]
```

Comment:
```suggestion
    #[error("could not complete Scan operation on DynamoDB; {0}")]
```

Reply:
Fixed in https://github.com/stacks-sbtc/sbtc/pull/1627/commits/afb291071eabe6f71779b09263438fba46094453